image/CMakeLists.txt

```diff
@@ -9,7 +9,11 @@ add_assets_target(
     SOURCE_DIR ${ASSETS_SOURCE_DIR}
     DEST_DIR ${ASSETS_DEST_DIR}
     DEVEL ${WITH_DEVELOPMENT_FEATURES}
-    DEPENDS json-target json-images-target
+    DEPENDS
+        json-common-target
+        "$<$<STREQUAL:${ASSETS},Proprietary>:json-proprietary-target>"
+        "$<$<STREQUAL:${ASSETS},Community>:json-community-target>"
+        "$<$<STREQUAL:${PROJECT_TARGET},TARGET_RT1051>:json-rt1051-target>"
```

Comment:
why do we need a flag for RT1051 nowhere? 
@swichu91 won't it be better to just include rt1051 in `only rt1051` cmakes?

Reply:
To exclude LUTS.bin from linux builds

Reply:
please see that other platform depended things have things like that done by including platform catalog. 

Now - these `JSON` targets are different per option. 

So instead making wonky ifs on strings in here. Why not define `json` target which will be defined based on options combinations to either:
json = json-common + proprietary + rt1051
json = json-common + oss + rt1051
json = json-common + oss
?

Reply:
well isnt like we are doing this already by defining target `assets`? All it does is to combine different jsons (like you mentioned into one target. How it would be different from the `assets` target?

Reply:
actually `add_assets_target(` calls our cmake function which does some magic, which creates targets and does some more magic. The easiest explanation I can find is:

In my opinion, it would be better to have the approach:
`if -> call to function ` 
I see it as the classic approach and it would mean to include target-dependent things, not add string comparisons in the function call.

Right now you are doing:
```
foo ( if ( something == propritetary) add_something (else if something == community) add_something_else && if (project_target == rt1051 ) add_rt1051)
```
where I think would make a bit more sense to:
```
list(actuall_dependencies)
# somewhere: ...

# 1: have a safe if's
if ( something == proprietary ) list(acctuall_dependencies append propritetary-json)
if ( something ==community) list(actuall_dependencies append community-json)

# 2: actually split PROJECT_TARGET dependent elements
if ( something != proprietary && something != community ) message(ERROR "This is fail safe: wrong $ASSETS_TARGET only proprietary and community available)
include ( ${PROJECT_TARGET}/assets)

# ... 

# then pass it to:
add_assets_target( .... DEPENDS actuall_dependencies)
```

In the end: yes. It's only about organizing the code here.  At best I would see: the removal hack of development assets too as a target, not wonky function call in `add_assets_target` and I know we have to live with what we have. I would just love if we could write it a bit more readable at least :) 

If you disagree. Then it's fine with me. I'm no expert in cmake and I do not feel this is the most important part of my review anyway :)